src/components/storage/aws.ts

```diff
@@ -24,41 +20,26 @@ export class AwsUserStorage implements UserStorage {
         this.bucketName = bucketName
       }
 
-    async storeUser(id:string, install_time: number) {
-        const key = 'users/users.csv'
-
-        const users = await this._getObject({key, type: 'csv'})
-        users.push([id, install_time])
-
-        let body;
-
-        await new Promise((resolve, reject) => {
-            stringify(users, function(err, output){
-                body = output;
-                resolve(output)
-            });
-        })
+    async storeUser(id:string, installTime: number) {
+        const key = 'users/' + id + '.json'
+        const body = {"id":id,"installTime":installTime}
 
-        await this._putObject({key, body, type: 'csv'})
-        return {id: id}
+        const success = await this._putObject({key, body, type: 'json'})
+        return {id, success}
     }
 
-    async isUserById(id) {
-        const key = 'users/users.csv'
-        const users = await this._getObject({key, type: 'csv'})
-
-        users.forEach((user) => {
-            if(user[0] == id) {
-                return false;
-            }
-        })
-        
-        return true;
+    async userExists(id) {
+        const key = 'users/' + id + '.json'
+        return await this._getObject({key, type: 'json'})        
     }
 
-    async _putObject({key, body, type, mime} : {key : string, body, type? : 'csv', mime? : string}) {
+    async _putObject({key, body, type, mime} : {key : string, body: any, type? : 'json', mime? : string}) {
```

Comment:
This type signature is repeated in both the user and event storage classes. Same issue with the `_getObject` methods. Immediate improvement could be to DRY it out by doing something like:

```ts
interface PutObjectProps {
  key: string
  body: any,
  type?: 'json'   // Please note that this is saying `type` field can only ever be set to the value 'json' or undefined!
  mime?: string
}
```

and changing your method to something like

```ts
async _putObject({key, body, type, mime}: PutObjectProps) {...}
```

Even better would be creating an abstract class that both these classes extend. The abstract class would have abstract `_putObject` and `_getObject` methods, which then these classes would be forced to implement. More info on abstract classes: https://www.typescriptlang.org/docs/handbook/classes.html#abstract-classes

Reply:
I totally overlooked this, but yes, they're almost the same implementation! You could put the shared logic into a method of a new class which both the event and user AWS storage classes extend. You should probably make that method `protected` (currently it's `public` as that is TS default for methods if you don't specify a privacy modifier).